### PR TITLE
Prepare publish of com.gradleup.shadow

### PR DESCRIPTION
**Pull request by Goooler**
- Plugin id from `com.github.johnrengelman.shadow` to `com.gradleup.shadow`.
- Support publishing to MavenCentral and with snapshots support.
- Add release workflows.
- Add changelog and releasing.
- Update original repo ref links.

### CODE REVIEW

**Comment by Goooler** [comment]
It's here https://oss.sonatype.org/content/repositories/snapshots/com/gradleup/shadow/.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 21
+      - uses: gradle/actions/setup-gradle@v3
+        with:
+          cache-read-only: true
+      # Disable CC due to https://github.com/gradle/gradle/issues/22779
+      - run: ./gradlew publish --no-configuration-cache
```
Comment (Goooler) [inline .github/workflows/release.yml:24]: If we still want to sync the plugin updates to GradlePluginPortal, need to add `publishPlugins` task here and provide

```sh
GRADLE_PUBLISH_KEY: ${{ secrets.GRADLE_PUBLISH_KEY }}
GRADLE_PUBLISH_SECRET: ${{ secrets.GRADLE_PUBLISH_SECRET }}
```

Comment (Goooler) [reply .github/workflows/release.yml:24]: May add GH pages task too, or in the next step with new deployment steps.
It's https://github.com/GradleUp/shadow/blob/main/gradle/ghPages.gradle now, but I'm unfamiliar with GH page deployment, can someone add related tasks or steps?

Comment (martinbonnin) [reply .github/workflows/release.yml:24]: Re: Gradle plugin portal, there's `GRADLE_KEY` and `GRADLE_SECRET` in the org. I think you should be able to see that from https://github.com/GradleUp/shadow/settings/secrets/actions. Let me know if now, we can add you there.

Re: GH pages, they're configured in the settings as well to deploy when pushes are made to `gh-pages`. I would say, maybe try to call `gitPublishCopy` manually?

Comment (Goooler) [reply .github/workflows/release.yml:24]: 1. https://github.com/GradleUp/shadow/settings/secrets/actions is 404 for me, may cause there is only maintainer permission for.

2. Yeah, seems we can update the env variable and call `gitPublishCopy`

https://github.com/GradleUp/shadow/blob/07f4e63a28b0c61fa9403d2e8320c6c5c0bcc678/gradle/ghPages.gradle#L6

Comment (martinbonnin) [reply .github/workflows/release.yml:24]: @Goooler I have made you an admin, should be much easier now!

Screenshot 2024-08-02 at 10 17 49

Comment (Goooler) [reply .github/workflows/release.yml:24]: See it now. Thanks!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 21
+      - uses: gradle/actions/setup-gradle@v4
+        with:
+          cache-read-only: true
+      # Disable CC due to https://github.com/gradle/gradle/issues/22779
+      - run: ./gradlew release --no-configuration-cache
```
Comment (Goooler) [inline .github/workflows/release.yml:24]: This `release` task is for publishing all things, and but I can't ensure `gitPublishPush` works well for website deployment, can someone give me a hand to test and fix it?


https://github.com/GradleUp/shadow/blob/b3ce1927c7cc1d1c5bc0847d086e39d749b96108/gradle/publish.gradle#L31-L33

**Comment by breandan** [comment]
Tried `id("com.gradleup.shadow") version "8.3.0-SNAPSHOT"`, but received the following error: 

```
* What went wrong:
Plugin [id: 'com.gradleup.shadow', version: '8.3.0-SNAPSHOT'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'com.gradleup.shadow:com.gradleup.shadow.gradle.plugin:8.3.0-SNAPSHOT')
  Searched in the following repositories:
    MavenRepo
    Gradle Central Plugin Repository

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
```

**Comment by Goooler** [comment]
@breandan See ff01bdee93d23bab376c041b89ec936d61fb9ac5.

File: .github/workflows/release.yml (hunk 2 above)
Comment (Goooler) [reply .github/workflows/release.yml:28]: 15780f5c5f2f295640f8724261d94b3844c56106

File: src/docs/about/README.md [hunk 3]
```diff
@@ -22,60 +22,6 @@ so Shadow was published there.
...
-* [Schalk W. Cronjé](https://github.com/ysb33r)
-* [Scott Newson](https://github.com/sgnewson)
-* [Serban Iordache](https://github.com/siordache)
-* [Sergey Tselovalnikov](https://github.com/SerCeMan)
-* [Tim Yates](https://github.com/timyates)
-* [Trask Stalnaker](https://github.com/trask)
-* [Tyler Benson](https://github.com/tylerbenson)
-* [Victor Tso](https://github.com/roxchkplusony)
-* [Yahor Berdnikau](https://github.com/Tapchicoma)
+<a href="https://github.com/GradleUp/shadow/graphs/contributors">
```
Comment (oleg-nenashev) [inline src/docs/about/README.md:25]: It works pretty well as long as Contributor equals to Code Committer
The governance charter has a wider definition: https://gradleup.com/docs/community/governance/#roles

I would advise changing the section title to "Committers", and then maybe add "Other Contributors"

File: gradle.properties [hunk 4]
```diff
@@ -1,4 +1,30 @@
 org.gradle.jvmargs=-Dfile.encoding=UTF-8 -Xmx4g -XX:MaxMetaspaceSize=2g
 org.gradle.parallel=true
 org.gradle.caching=true
-org.gradle.configuration-cache=true
\ No newline at end of file
+org.gradle.configuration-cache=true
+
+
+GROUP=com.gradleup.shadow
```
Comment (oleg-nenashev) [inline gradle.properties:7]: Maybe `com.gradleup.plugins`? Started the discussion on Slack

File: src/docs/about/README.md (hunk 3 above)
Comment (Goooler) [reply src/docs/about/README.md:25]: I don't think it should be Committers or something, these pictures and links are linked to https://github.com/GradleUp/shadow/graphs/contributors, they are all named `Contributors` by GitHub.

File: gradle.properties (hunk 4 above)
Comment (Goooler) [reply gradle.properties:7]: Borrowed from https://github.com/GradleUp/nmcp/blob/b6325eac2ca6111a8d22a26be401d021fda90003/build.gradle.kts#L23.

Comment (martinbonnin) [reply gradle.properties:7]: I like `com.gradle.shadow`

It allows to easily make a major version bumps if needed (see https://jakewharton.com/java-interoperability-policy-for-major-version-updates/). Also groups all the artifacts together and is short enough.

File: src/docs/about/README.md (hunk 3 above)
Comment (oleg-nenashev) [reply src/docs/about/README.md:25]: Wearing my community manager hat, that belongs to "GitHub: Bad Parts" unfortunately

**Review by oleg-nenashev — APPROVED** [review]
To confirm, none of my comments is a blocking one

File: src/docs/about/README.md (hunk 3 above)
Comment (martinbonnin) [reply src/docs/about/README.md:25]: Having a single source of truth in GitHub is quite handy. If someone makes a non-code contribution, we could ask them to make a small commit in the repo to acknowledge to have them listed.

**Pull request merged**